app/src/main/java/org/kinecosystem/kinit/view/customView/AnswerSelectedOverView.kt

```diff
@@ -0,0 +1,59 @@
+package org.kinecosystem.kinit.view.customView
+
+import android.content.Context
+import android.os.Build
+import android.support.constraint.ConstraintLayout
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.MotionEvent
+import android.view.ViewAnimationUtils
+import androidx.core.animation.doOnEnd
+import org.kinecosystem.kinit.R
+
+
+class AnswerSelectedOverView : ConstraintLayout {
+
+
+    interface OnSelectionListener {
+        fun onSelected()
+    }
+
+    var listener: OnSelectionListener? = null
+
+    constructor(context: Context) : super(context) {
+        init()
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        init()
+    }
+
+    private fun init() {
+        val inflater = context
+                .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+        inflater.inflate(R.layout.answer_image_selected_view, this, true)
+        setOnTouchListener { _, event ->
```

Comment:
you could pass the index of the answer to the view as an argument and then you won't have to have an array of listeners inside the model. instead you have a single listener that takes the index as a parameter